### PR TITLE
fix(collector): fix dns collector limited to 63 chars

### PR DESCRIPTION
**Pull request by DexterYan**
## Description, Motivation and Context

- validation.IsQualifiedName is using IsDNS1123Subdomain which has 63 limit based on RFC 1123
- change to use Hostname RFC 952 if it less 255

[sc-115977](https://app.shortcut.com/replicated/story/115977/dns-collector-limited-to-63-chars)

## Checklist

- [x] New and existing tests pass locally with introduced changes.
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] The commit message(s) are informative and highlight any breaking changes
- [ ] Any documentation required has been added/updated. For changes to https://troubleshoot.sh/ create a PR [here](https://github.com/replicatedhq/troubleshoot.sh/pulls)

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: pkg/collect/host_network_test.go [hunk 1]
```diff
@@ -44,8 +44,13 @@ func Test_isValidLoadBalancerAddress(t *testing.T) {
 			want: false,
 		},
 		{
-			name: "Too many characters",
+			name: "Too many characters less than 255",
```
Comment (xavpaice) [inline pkg/collect/host_network_test.go]: what is this testing for?  The name needs to be really clear.

File: pkg/collect/host_network_test.go [hunk 2]
```diff
@@ -44,8 +44,13 @@ func Test_isValidLoadBalancerAddress(t *testing.T) {
...
 			args: args{address: "howlongcanwemakethiswithoutrunningoutofwordsbecasueweneedtohitatleast64.com:80"},
+			want: true,
+		},
+		{
+			name: "Too many characters bigger than 255",
```
Comment (xavpaice) [inline pkg/collect/host_network_test.go]: let's name this something more clear for what we're testing

**Review by xavpaice — CHANGES_REQUESTED** [review]
Just some really minor nits on the wording in tests, change LGTM otherwise

File: pkg/collect/host_network_test.go (hunk 1 above)
Comment (DexterYan) [reply pkg/collect/host_network_test.go]: good catch. I have updated those test names

**Review by xavpaice — APPROVED** [review]

**Pull request merged**